hrjobcontract/js/test/controllers/modal/modal-contract_test.js

```diff
@@ -0,0 +1,146 @@
+          period_end_date: null,
+          end_reason: "1",
+          notice_amount: null,
+          notice_unit: null,
+          notice_amount_employee: null,
+          notice_unit_employee: null,
+          location: "Headquarters",
+          jobcontract_revision_id: "60"
+        },
+        hour: {},
+        pay: {},
+        leave: [],
+        health: {},
+        pension: {}
+      };
+
+      // $uibModalInstanceSpy for $uibModalInstance.opened.then()
+      $uibModalInstanceSpy = {
+        opened: {
+          then: function() {
+            jasmine.createSpy('modalInstance.opened');
+          }
+        }
+      };
+
+      // $uibModalSpy for $modal.open()
+      $uibModalSpy = {
+        open: function() {
+          return {
+            result: {
+              then: function(callback) {
+                callback("edit");
+              }
+            }
+          }
+        }
+      };
+
+      $httpBackend.whenGET(/action=validatedates&entity=HRJobDetails/).respond({ success: true });
+      $httpBackend.whenGET(/action=get&entity=HRJobContract/).respond({});
+      $httpBackend.whenPOST(/action=create&entity/).respond({});
+      $httpBackend.whenPOST(/action=replace&entity/).respond({});
+      $httpBackend.whenGET(/views.*/).respond({});
+
+      makeController();
+    }));
+
+
+    describe("Sets contract end date as appropriate", function() {
+
+      beforeEach(function() {
+        createModalSpy();
+        $scope.save();
+        $rootScope.$digest();
+      });
+
+      describe("When period_end_date is null", function() {
+
+        it("Sets contract period_end_date to '' ", function() {
+          expect($scope.entity.details.period_end_date).toBe("");
+        });
+      });
+
+      describe("When period_end_date is not null", function() {
+
+        beforeEach(function() {
+          $scope.entity.details.period_end_date = '02-03-2017';
+        });
+
+        it("Sets contract period_end_date to '02-03-2017' ", function() {
+          expect($scope.entity.details.period_end_date).toBe('02-03-2017');
+        });
+      });
+    });
+
+    function makeController() {
+      $scope = $rootScope.$new();
+      $scope.allowSave = true;
+      $scope.is_primary = '';
+
+      ctrl = $controller('ModalContractCtrl', {
+        $scope: $scope,
+        $uibModal: $uibModalSpy,
+        $uibModalInstance: $uibModalInstanceSpy,
+        ContractDetailsService: ContractDetailsService,
+        action: 'edit',
+        entity: entity,
+        content: {
+          allowSave: true
+        },
+        files: {},
+        utils: {
+          contractListLen: 1
+        }
+      });
+    }
+
+    function createModalSpy() {
+      spyOn(ContractDetailsService, "validateDates").and.callFake(function() {
+        var deferred = $q.defer();
+        deferred.resolve({
+          success: true
+        });
+
+        return deferred.promise;
+      });
+
+      spyOn($uibModalSpy, "open").and.callThrough();
```

Comment:
Also what is the purpose of spying? If you does not need to test if it is called.